packages/next/client/image.tsx

```diff
@@ -276,6 +292,40 @@ export default function Image({
   if (!configEnableBlurryPlaceholder) {
     placeholder = 'empty'
   }
+  const isStatic = typeof src === 'object'
+  let staticSrc = ''
+  if (isStatic) {
+    let staticImport = src as StaticImport
+    let staticImageData: StaticImageData
+    if ((staticImport as StaticRequire).default) {
+      staticImageData = (staticImport as StaticRequire).default
+    } else {
+      staticImageData = staticImport as StaticImageData
+    }
```

Comment:
```suggestion
  if (typeof src === 'object') {
    let staticImageData: StaticImageData
    if ('default' in src) {
      staticImageData = src.default
    } else {
      staticImageData = src
    }
```

This will avoid casting

Reply:
I got rid of these casts by implementing some type guards for these types.